web3/_utils/module_testing/eth_module.py

```diff
@@ -4149,6 +4149,36 @@ def test_eth_estimate_gas_with_block(
         assert is_integer(gas_estimate)
         assert gas_estimate > 0
 
+    @pytest.mark.parametrize(
+        "params",
+        (
+            {
+                "nonce": 1,  # int
+                "balance": 1,  # int
+                "code": HexStr("0x"),  # HexStr
+                # with state
+                "state": {HexStr(f"0x{'00' * 32}"): HexStr(f"0x{'00' * 32}")},
+            },
+            {
+                "nonce": HexStr("0x1"),  # HexStr
+                "balance": HexStr("0x1"),  # HexStr
+                "code": b"\x00",  # bytes
+                # with stateDiff
+                "stateDiff": {HexStr(f"0x{'00' * 32}"): HexStr(f"0x{'00' * 32}")},
+            },
+        ),
+    )
+    def test_eth_estimate_gas_with_override_param_type_check(
```

Comment:
We added the async functionality in this PR, let's add an analogous async test for this too. I noticed the `eth_call` equivalent also doesn't have an async test... it would be nice to add that as well and it could be in this PR if not too much of a burden.